accepted/0000-automatic-migration-salt-bundle.md

```diff
@@ -0,0 +1,81 @@
+- The `mgr_avoid_venv_salt_minion` pillar is not `True` for this minion.
+
+If so, the migration is performed in the following steps:
+
+- Install the `venv-salt-minion` package.
+- Copy the minion id file, minion configuration files, PKI files and static grains from the existing `salt-minion` in `/etc/salt` to the `venv-salt-minion` corresponding directories in `/etc/venv-salt-minion`.
+- Enable the `venv-salt-minion` service.
+- Disable the `salt-minion` service.
+- Remove `/etc/salt/minion.d/susemanager.conf` file, to prevent possible accidental reconnection of classic "salt-minion" after migration.
+
+The whole process is transparent to the user.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- The `salt-minion` package remains installed and the configuration files in the `/etc/salt` are not deleted.
+
+- After migration this check is done in every highstate.
+
+- If a client was bootstrapped avoiding the Salt Bundle following the steps in the documentation, the migration would force this client to use Salt Bundle. The options to avoid installing Salt Bundle and keep using `salt-minion` that currently appear in the documentation are:
+
+  * Execute mgr-bootstrap with --no-bundle option.
+
+  * Set AVOID_VENV_SALT_MINION to 1 in the generated bootstrap script.
+
+  * For bootstrap state set the `mgr_avoid_venv_salt_minion` pillar to `True`.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Force the removal of the `salt-minion` package and related files. This could be done by means of a custom grain module that informs if the minion is running  `venv-salt-minion`. With that in place, the application of the highstate a second time could handle that second stage for purging.
+- The impact of not doing this is that Salt 3000 clients will stop receiving updates for the Salt package.
+
+
+# Resolved questions
+[resolved]: #resolved-questions
+
+1. What is a better solution?
+  - The sls contains the logic for triggering the migration or not.
+  - There is logic for triggering the migration that decides if adding or not the sls to the highstate.
+  
+The option chosen is adding the logic to the state as this stays in line with our usage of SLS files.
+
+2. Could this automatic migration behaviour be opted out? It seems it would be reasonable in case bootstrapping with the Salt Bundle was explicitly avoided.
+
+The automatic migration can be opted out by setting the pillar `mgr_avoid_venv_salt_minion` to `True` for the specific minion.
+
+3. Is it best to take a cleaner approach and remove the `salt-minion` package and related files or to take a more conservative approach and do not do it in case there are unexpected scenarios?
+
+The approach that is going to be followed is removing the `salt-minion` package in the second stage in order to avoid the user from startig the `salt-minion` service manually. Configuration files in `/etc/salt` are not going to be removed in case the user had configured another `salt-minion` between stages.
```

Comment:
Is this sentence correct? My understanding was that we are NOT removing the "salt-minion" package as part of this migration.